modules/library/data/src/main/java/org/geotools/data/store/ContentFeatureSource.java

```diff
@@ -425,6 +427,35 @@ public final ReferencedEnvelope getBounds(Query query) throws IOException {
         } else {
             bounds = getBoundsInternal(query);
         }
+        // reprojection
+        if ( !canReproject() ) {
+            CoordinateReferenceSystem sourceCRS = query.getCoordinateSystem();
+            CoordinateReferenceSystem targetCRS = query.getCoordinateSystemReproject();
+            CoordinateReferenceSystem nativeCRS = getSchema().getCoordinateReferenceSystem();
+            
+            if (sourceCRS != null && !sourceCRS.equals(nativeCRS)) {
+                //override native crs
+                bounds = new ReferencedEnvelope(bounds, sourceCRS);
+            } else {
+                //no override
+                sourceCRS = nativeCRS;
+            }
+            if (targetCRS != null) {
+                
+                if(sourceCRS == null) {
+                    throw new IOException("Cannot reproject data, the source CRS is not available");
+                } else if(!sourceCRS.equals(targetCRS)) {
+                    try {
+                        bounds = bounds.transform(targetCRS, true);
+                    } catch (Exception e) {
+                        if(e instanceof IOException)
+                            throw (IOException) e;
+                        else
+                            throw (IOException) new IOException("Error occurred trying to reproject data").initCause(e);
```

Comment:
Yeah, I just used the same structure and wording as in the reprojection section of getFeatures() for consistency.